src/phoenix/experimental/callbacks/llama_index_trace_callback_handler.py

```diff
@@ -106,10 +106,16 @@ def payload_to_semantic_attributes(
     if EventPayload.PROMPT in payload:
         ...
     if EventPayload.MESSAGES in payload:
-        attributes[LLM_MESSAGES] = [
-            _message_payload_to_attributes(message_data)
-            for message_data in payload[EventPayload.MESSAGES]
-        ]
+        messages = payload[EventPayload.MESSAGES]
+        # Messages is only relevant to the LLM invocation
+        if event_type is CBEventType.LLM:
+            attributes[LLM_MESSAGES] = [
+                _message_payload_to_attributes(message_data) for message_data in messages
+            ]
+        elif event_type is CBEventType.AGENT_STEP and len(messages):
+            # the agent step contains a message that is actually the input
+            # akin to the query_str
+            attributes[INPUT_VALUE] = _message_payload_to_str(messages[0])
```

Comment:
Are we sure there's only one message here?

Reply:
For simple agent rooted spans yes. But can double check